TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/CapturedTrafficToHttpTransactionAccumulator.java

```diff
@@ -175,103 +171,139 @@ private static enum CONNECTION_STATUS {
                 requestCounter.incrementAndGet();
             }
-            log.atTrace().setMessage(()->"Adding request data for accum[" + connectionId + "]=" + accum).log();
+            log.atTrace().setMessage(() -> "Adding request data for accum[" + connectionId + "]=" + accum).log();
             accum.rrPair.addRequestData(timestamp, observation.getRead().getData().toByteArray());
-            log.atTrace().setMessage(()->"Added request data for accum[" + connectionId + "]=" + accum).log();
-        } else if (observation.hasWrite()) {
-            assert accum != null && accum.state == Accumulation.State.REQUEST_SENT;
-            var runningList = accum.rrPair;
-            if (runningList == null) {
-                throw new RuntimeException("Apparent out of order exception - " +
-                        "found a purported write to a socket before a read!");
-            }
-            log.atTrace().setMessage(()->"Adding response data for accum[" + connectionId + "]=" + accum).log();
-            runningList.addResponseData(timestamp, observation.getWrite().getData().toByteArray());
-            log.atTrace().setMessage(()->"Added response data for accum[" + connectionId + "]=" + accum).log();
+            log.atTrace().setMessage(() -> "Added request data for accum[" + connectionId + "]=" + accum).log();
         } else if (observation.hasEndOfMessageIndicator()) {
             handleEndOfRequest(accum);
         } else if (observation.hasReadSegment()) {
-            rotateAccumulationOnReadIfNecessary(connectionId, accum);
-            assert accum.state == Accumulation.State.NOTHING_SENT;
+            assert accum.state == Accumulation.State.ACCUMULATING_READS;
             log.atTrace().setMessage(()->"Adding request segment for accum[" + connectionId + "]=" + accum).log();
             if (accum.rrPair.requestData == null) {
                 accum.rrPair.requestData = new HttpMessageAndTimestamp.Request(timestamp);
                 requestCounter.incrementAndGet();
             }
             accum.rrPair.requestData.addSegment(observation.getReadSegment().getData().toByteArray());
             log.atTrace().setMessage(()->"Added request segment for accum[" + connectionId + "]=" + accum).log();
+        } else if (observation.hasSegmentEnd()) {
+            assert accum.rrPair.requestData.hasInProgressSegment();
+            accum.rrPair.requestData.finalizeRequestSegments(timestamp);
+        }
+        return Optional.of(CONNECTION_STATUS.ALIVE);
+    }
+
+    private Optional<CONNECTION_STATUS> handleObservationForWriteState(Accumulation accum,
+                                                                       TrafficObservation observation,
+                                                                       String connectionId,
+                                                                       Instant timestamp) {
+        if (accum.state != Accumulation.State.ACCUMULATING_WRITES) {
```

Comment:
This is defensive coding.  handleObservationForReadState() needs to do it since we fall through (actually, to this state).  There is logging at the top level if we come out of the chain without handling the observation (see [line 185](https://github.com/opensearch-project/opensearch-migrations/pull/368/files/9a7139fae08caa69b2e6322a09e671e41bc3ba5e#diff-c11d4821bd721ac56c8c6d0a1f58eec12894b9c17e7c0a45a94465c328d01156R186)